addons/account/models/account_move.py

```diff
@@ -4123,6 +4123,16 @@ def button_draft(self):
         if any(move.state not in ('cancel', 'posted') for move in self):
             raise UserError(_("Only posted/cancelled journal entries can be reset to draft."))
 
+        self._checks_action_draft()
+
+        for move in self:
+            # We remove all the analytics entries for this journal
+            move.mapped('line_ids.analytic_line_ids').unlink()
```

Comment:
I believe you can directly go for 
```suggestion
        self.mapped('line_ids.analytic_line_ids').unlink()
```
and make a single block of four lines (no empty lines from 4126 to 4133)